src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/media/index.js

```diff
@@ -0,0 +1,65 @@
+import pick from 'ramda/src/pick';
+
+const generateVideoBlock = block => {
+  const generatedBlock = {
+    type: 'aresMediaMetadata',
+    blockId: `urn:bbc:ares::${block.subType}:${block.id}`,
+    model: pick(
+      [
+        'advertising',
+        'embedding',
+        'format',
+        'id',
+        'imageCopyright',
+        'imageUrl',
+        'subType',
+        'synopses',
+        'title',
+        'versions',
+      ],
+      block,
+    ),
+  };
+
+  if (generatedBlock.model.format === 'video') {
+    generatedBlock.model.format = 'audio_video';
+  }
+
+  return generatedBlock;
+};
+
+const generateImageBlock = block => {
+  return {
+    type: 'image',
+    model: {
+      blocks: [
+        {
+          type: 'rawImage',
+          model: {
+            copyrightHolder: block.imageCopyright,
+            locator: block.imageUrl,
+          },
+        },
+      ],
+    },
+  };
+};
+
+const convertMedia = block => {
+  return {
+    type: 'video',
+    model: {
+      locator: `urn:bbc:pips:pid:${block.id}`,
+      blocks: [
+        {
+          type: 'aresMedia',
+          model: {
+            blocks: [generateVideoBlock(block), generateImageBlock(block)],
+          },
+        },
+      ],
+    },
+  };
+};
+
+export default convertMedia;
```

Comment:
Not necessarily here, but would it be worth carrying out some sort of validation of the CPS media block and fail to return an `aresMedia` block at all if it is incomplete/malformed? Otherwise aren't we running the risk of outputting blocks containing a bunch of `undefined`s if we encounter exotic CPS assets (relates to comment below as well)?

Reply:
Episodes are generally not used in CPS because they aren't supported by Trevor and therefore Apps (currently).  Current Web does support them though, so we should.  I don't think I have enough context at the moment to comment further.

Reply:
Added a validator that will prevent the block from rendering if it seems to have junk value